tendermint/src/time.rs

```diff
@@ -16,33 +18,35 @@ use crate::error::Error;
 /// Tendermint timestamps
 /// <https://github.com/tendermint/spec/blob/d46cd7f573a2c6a2399fcab2cde981330aa63f37/spec/core/data_structures.md#time>
 #[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Serialize, Deserialize)]
-#[serde(try_from = "Timestamp", into = "Timestamp")]
-pub struct Time(DateTime<Utc>);
+#[serde(try_from = "Timestamp")]
+pub struct Time(pub DateTime<Utc>);
 
 impl Protobuf<Timestamp> for Time {}
 
-impl From<Timestamp> for Time {
-    fn from(value: Timestamp) -> Self {
-        // prost_types::Timestamp has a SystemTime converter but
-        // tendermint_proto::Timestamp can be JSON-encoded
-        let prost_value = prost_types::Timestamp {
-            seconds: value.seconds,
-            nanos: value.nanos,
-        };
+impl TryFrom<Timestamp> for Time {
+    type Error = Error;
 
-        SystemTime::from(prost_value).into()
+    fn try_from(value: Timestamp) -> Result<Self, Error> {
+        let nanos = value.nanos.try_into().map_err(Error::timestamp_overflow)?;
+
+        Ok(Time(Utc.timestamp(value.seconds, nanos)))
     }
 }
 
-impl From<Time> for Timestamp {
-    fn from(value: Time) -> Self {
-        // prost_types::Timestamp has a SystemTime converter but
-        // tendermint_proto::Timestamp can be JSON-encoded
-        let prost_value = prost_types::Timestamp::from(SystemTime::from(value));
```

Comment:
I remove the use of `SystemTime` here, as this is the only place where the `tendermint` crate is still using `std`.